Sps30 multithreading

Lilypads opened the pull request:
add comments and documentations

Lilypads committed:
restructuring and cleaning

Lilypads committed:
minor fix

Lilypads committed:
print serial number as string

BenjaminFrazer commented on src/sps30demo.cpp:
```diff
@@ -49,11 +37,18 @@ int main()
+SPS30settings s;
+s.autoStartThread = false;
 
+mySPS.startMeasurement(s);
```
I don't like that we have to pass the settings in only during start measurement... this makes testing individual methods without hardware very difficult... what if the user doesn't want to run continuous data acquisition? Currently we are forcing the user to start acquisition in order to set the device config. I suggest, we move the settings back into the constructor, and add an optional method setSettings(sps30Settings).

BenjaminFrazer reviewed (changes requested):
move the settings back into the constructor (optionally) and then define a setter that also takes the settings struct

BenjaminFrazer commented:
Also sps30lib.h (a reference library) is still in the the top level src directory

BenjaminFrazer committed:
+settings method, -settings from startMeas()

BenjaminFrazer committed:
Merge branch 'main'

Conflicts:
	src/libsps30.cpp
	src/try1.cpp

BenjaminFrazer committed:
Updatated test workflow

explicitly followed instructions on https://abyz.me.uk/rpi/pigpio/download.html

BenjaminFrazer commented:
merged with restructuring branch which will then get merged into main
